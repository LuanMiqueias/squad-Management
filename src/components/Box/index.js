import React from "react";
import styles from "./style.module.css";
import iconPlus from "../../assets/icons/add.svg";

const Box = ({
  title,
  isAddItems,
  children,
  isBackground,
  onClick,
  classNameBox,
}) => {
  return (
    <div
      className={`${classNameBox} ${styles.box} ${
        isBackground && styles.backgroundBox
      } `}
    >
      <div className={title && styles.title}>
        <h1>{title}</h1>
        {isAddItems && (
          <button
            className={styles.addIcon}
            onClick={() => onClick()}
            aria-label="Create new team"
          >
            <img src={iconPlus} alt="" />
          </button>
        )}
      </div>
      <div className={styles.boxContent}>{children}</div>
    </div>
  );
};

export default Box;
