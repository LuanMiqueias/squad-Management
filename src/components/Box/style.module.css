.box{
  width: 100%;
  max-width: 100%;
  background: var(--white);
  border-radius: 10px;
  box-shadow: 0 0 5px rgba(0, 0, 0, .05);
}
.backgroundBox{
  background: var(--gradient-background-box);
}
.title{
  padding: 24px 24px 24px 32px;
  border-bottom: 1px solid var(--divider);
  display: flex;
  align-items: center;
  justify-content: space-between;
}
.addIcon{
  width: 35px;
  height: 35px;
  background: var(--button-gradient);
  box-shadow: 0px 5px 10px 1px rgba(208, 60, 108, .35);
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  color:#fff;
  outline: none;
  border:none;
  transition: all ease .1s;
}
.addIcon:hover{
  filter: hue-rotate(-5deg);
  transform: scale(1.1);
}
.boxContent{
  width: 100%;
  display: flex;
  justify-content: center;
  gap:32px;
}