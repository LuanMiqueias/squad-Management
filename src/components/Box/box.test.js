import { render } from "@testing-library/react";
import Box from "./";

it("Shold math snapshot", () => {
  const { container } = render(<Box title="test" classNameBox="classTest" />);

  expect(container.firstChild).toMatchSnapshot();
});

it("Should background if isBackground to equal true", () => {
  const { container } = render(
    <Box title="test" classNameBox="classTest" isBackground={true} />
  );
  expect(container.firstChild).toHaveClass("backgroundBox");
});

it("Should not have background if isBackground to equal false", () => {
  const { container } = render(
    <Box title="test" classNameBox="classTest" isBackground={false} />
  );
  expect(container.firstChild).not.toHaveClass("backgroundBox");
});

it("Should have button add-team if isAddItems to equal true", () => {
  const { queryByLabelText } = render(
    <Box title="test" classNameBox="classTest" isAddItems={true} />
  );
  const textElement = queryByLabelText(/create new team/i);
  expect(textElement).toBeInTheDocument();
});

it("Should not have button add-team if isAddItems to equal false", () => {
  const { queryByLabelText } = render(
    <Box title="test" classNameBox="classTest" isAddItems={false} />
  );
  const textElement = queryByLabelText(/create new team/i);
  expect(textElement).not.toBeInTheDocument();
});
