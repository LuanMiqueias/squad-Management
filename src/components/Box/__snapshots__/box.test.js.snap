// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shold math snapshot 1`] = `
<div
  class="classTest box undefined "
>
  <div
    class="title"
  >
    <h1>
      test
    </h1>
  </div>
  <div
    class="boxContent"
  />
</div>
`;
